examples/seismic/skew_self_adjoint/operators.py

```diff
@@ -0,0 +1,333 @@
+# TODO figure out if can get time sampling from the SparseTimeFuntions src/rec
+# TODO replace the conditional logic in the stencil with comprehension
+from devito import Eq, Operator, Function, TimeFunction
+
+
+def iso_stencil(field, model, **kwargs):
+    """
+    Stencil for the scalar isotropic visco- acoustic variable density
+    skew self adjoint wave equation:
+
+        b/v^2 [ P.dt2 + w/Q P.dt ] = (b P.dx).dx + (b P.dy).dy + (b P.dz).dz + s
+
+    Note derivative shifts are omitted for simplicity above.
+    See implementation notebook ssa_01_iso_implementation1.ipynb for more details.
+
+    Parameters
+    ----------
+    field : TimeFunction, required
+        The pressure wavefield computed solution.
+    model : Dictionary <string>:<Function>, contains:
+        'b': Buoyancy = reciprocal density (units: m^3/kg)
+        'v': Velocity (units: m/msec or km/sec)
+        'wOverQ': The w/Q field for dissipation only attenuation.
+    forward : bool, optional
+        The propagation direction. Defaults to True.
+    q : TimeFunction, Function or float, optional
+        Full-space/time source of the wave-equation.
+
+    Returns
+    ----------
+    The time update stencil.
+    """
+    # Get the Functions for buoyancy, velocity, and wOverQ
+    assert 'b' in model, "model dictionary must contain key 'b'"
+    assert 'v' in model, "model dictionary must contain key 'b'"
+    assert 'wOverQ' in model, "model dictionary must contain key 'wOverQ'"
+    b = model['b']
+    v = model['v']
+    wOverQ = model['wOverQ']
+
+    # Define time step of pressure wavefield to be updated
+    forward = kwargs.get('forward', True)
+
+    if forward:
+        field_next = field.forward
+        field_prev = field.backward
+    else:
+        field_next = field.backward
+        field_prev = field.forward
+
+    # Get the source
+    q = kwargs.get('q', 0)
+
+    # Define the time update equation for 2d/3d
+    # TODO: consider replacing conditional logic with @mloubout's comprehension
+    # space_fd = sum([getattr(b * getattr(field, 'd%s'%d.name)(x0=d+d.spacing/2)),
+    #            'd%s'%d.name)(x0=d-d.spacing/2)) for d in field.dimensions[1:]])
+    if len(field.data.shape) == 3:
+        t, x, z = field.dimensions
+        eq_time_update = (t.spacing**2 * v**2 / b) * \
+            ((b * field.dx(x0=x+x.spacing/2)).dx(x0=x-x.spacing/2) +
+             (b * field.dz(x0=z+z.spacing/2)).dz(x0=z-z.spacing/2) + q) + \
+            (2 - t.spacing * wOverQ) * field + \
+            (t.spacing * wOverQ - 1) * field_prev
+
+    else:
+        t, x, y, z = field.dimensions
+        eq_time_update = (t.spacing**2 * v**2 / b) * \
+            ((b * field.dx(x0=x+x.spacing/2)).dx(x0=x-x.spacing/2) +
```

Comment:
@FabioLuporini this is what the `getattr` stuff does but for generic number of dimensions